tritium-metrics/src/main/java/com/palantir/tritium/metrics/executor/TaggedMetricsExecutorService.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.metrics.executor;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import com.palantir.tritium.metrics.registry.MetricName;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public final class TaggedMetricsExecutorService implements ExecutorService {
+
+    private final ExecutorService delegate;
+
+    private final Meter submitted;
+    private final Counter running;
+    private final Meter completed;
+    private final Timer duration;
+    private final Timer queuedDuration;
+
+    public TaggedMetricsExecutorService(
+            ExecutorService delegate,
+            TaggedMetricRegistry registry,
+            Map<String, String> safeTags) {
+        this.delegate = delegate;
```

Comment:
probably want to check that safeTags is non-null & not-empty so that they have reasonable metric names
```suggestion
        this.delegate = checkNotNull(delegate, "delegate");
        checkArgument(!safeTags.isEmpty(), "safeTags cannot be empty");
```